app/smc/pytest/e2e_smoke.py

```diff
@@ -583,3 +599,59 @@ def test_aiclk(arc_chip_dut, asic_id):
         aiclk = arc_chip.arc_msg(ARC_MSG_TYPE_GET_AICLK)[0]
         assert aiclk == clk, f"Failed to set clock to {clk} MHz"
         logger.info(f"AICLK set to {aiclk} MHz successfully")
+
+
+def upgrade_from_version_test(tmp_path: Path, board_name, unlaunched_dut, version):
```

Comment:
Generally looks good- a few things I might add:
- let's validate that the DMFW we are running matches the expected version (we can pull this from telemetry). That way we know that the update completed correctly. We can check CMFW version as well in telemetry.
- we could run this test first by defining it first in the file. If we do that we may be able to skip flashing/launching the DUT later, and improve test execution times by only flashing main once.